--- Xdmf/libsrc/XdmfArray.h.orig	2010-03-24 21:03:48.000000000 +0100
+++ Xdmf/libsrc/XdmfArray.h	2013-09-13 16:09:37.644518551 +0200
@@ -46,6 +46,7 @@ accomplished directly thru XdmfArrays by
 
 #ifndef SWIG
 #include <stdlib.h>
+#include <string.h>
 #endif
 
 class XDMF_EXPORT XdmfArray : public XdmfDataDesc {
--- Xdmf/libsrc/XdmfObject.h.orig	2009-12-23 18:23:49.000000000 +0100
+++ Xdmf/libsrc/XdmfObject.h	2013-09-13 16:13:03.814509766 +0200
@@ -81,6 +81,7 @@
 
 # ifdef UNIX
 #  include "sys/file.h"
+#  include <string.h>
 #  include "strings.h"
 #  define STRCASECMP strcasecmp
 #  define STRNCASECMP strncasecmp
--- Xdmf/libsrc/utils/XdmfExodusReader.cxx.orig	2010-03-18 16:24:25.000000000 +0100
+++ Xdmf/libsrc/utils/XdmfExodusReader.cxx	2013-09-13 17:25:22.130994452 +0200
@@ -24,6 +24,7 @@
 /*******************************************************************/
 
 #include "XdmfExodusReader.h"
+#include <algorithm>
 #include <exodusII.h>
 
 //
